### ISSUE TITLE
ItemSend for shared mailboxes blocking emails from being sent

### ISSUE THREAD

**Issue by elemen7s** [OfficeDev/office-js#3527]
When sending an email from a shared mailbox, the email is not sent and is blocked because the event ItemSend is still being hit even though SupportsSharedFolders is not set in the manifest and the default should be false.

The event.completed method below is getting ignored too on ItemSend.
`event.completed({ allowEvent: true });`

Office.context.mailbox.initialData contains an error code of 9020 and permissionLevel of 0 as below: 
- error:1 
- errorCode: 9020
- permissionLevel: 0

The same addin works fine for a user mailbox.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: 
* Operating System: Windows 11
* Browser (if using Office on the web): All browsers
 
## Expected behavior
Addin events should not be hit if SupportsSharedFolders is not specified and set to true in the manifest

## Current behavior
Shared mailboxes are blocked from sending any emails

## Steps to reproduce
Create an addin with ItemSend event which does nothing except to call event.completed({ allowEvent: true })
Send an email from a shared mailbox

**Comment by exextoc** [comment]
Hey @elemen7s 

It's not a repro for us. Can you please share a video or a sample add-in which can help us investigate it further ?

**Comment by elemen7s** [comment]
Uploaded here - https://wetransfer.com/downloads/25c01a6d6c04df5961a3cde7894d4bc620230721113828/52cf66

**Comment by exextoc** [comment]
Hi @elemen7s, thanks for sharing the video. From the video it seems like the email is being sent from the context of a group and not a shared mailbox as specified in the title of this issue, Could you please confirm?

**Comment by elemen7s** [comment]
Yes that is correct. At the time of opening this issue the same exact issue was happening when sending a mail using the 'Send email' button marked in below screenshot, but it seems like this particular case has been fixed. 

Only the issue shown in the video remains.

![image](https://github.com/OfficeDev/office-js/assets/10210686/5e213a21-a942-4739-bc72-4306a95ff405)

**Comment by exextoc** [comment]
Thanks for confirming. We are able to repro this issue and it has been put into our backlog. We unfortunately have no timelines to share at this point. Thanks!

Internal tracking id: 146213
